pkgs/os-specific/linux/nvidia-x11/settings.nix

```diff
@@ -104,19 +116,17 @@ stdenv.mkDerivation {
 
   nativeBuildInputs = [ pkg-config m4 addOpenGLRunpath ];
 
-  buildInputs = [ jansson libXv libXrandr libXext libXxf86vm libvdpau nvidia_x11 gtk2 dbus ]
+  buildInputs = [ jansson libXv libXrandr libXext libXxf86vm libvdpau nvidia_x11 dbus ]
+    ++ lib.optionals (withGtk2 || lib.versionOlder nvidia_x11.settingsVersion "525.53") [ gtk2 ]
     ++ lib.optionals withGtk3 [ gtk3 librsvg wrapGAppsHook3 ];
```

Comment:
If we're fixing this package anyway then `wrapGAppsHook3` belongs in `nativeBuildInputs`. otherwise it looks good to me